core/src/main/java/org/elasticsearch/search/rescore/QueryRescorer.java

```diff
@@ -35,15 +36,9 @@
 public final class QueryRescorer implements Rescorer {
 
     public static final Rescorer INSTANCE = new QueryRescorer();
-    public static final String NAME = "query";
 
     @Override
-    public String name() {
-        return NAME;
-    }
-
-    @Override
-    public TopDocs rescore(TopDocs topDocs, SearchContext context, RescoreSearchContext rescoreContext) throws IOException {
+    public TopDocs rescore(TopDocs topDocs, IndexSearcher searcher, RescoreContext rescoreContext) throws IOException {
```

Comment:
Not really too sensitive, just pretty likely to break in the future, I guess. I'd be happy to include the original query somehow (just give it the `Query`, `SearchSourceBuilder`?) if it is important though I'm not sure it is important to rescorers to know the original query.